pkg/router/controller/route_secret_manager_test.go

```diff
@@ -0,0 +1,1139 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/openshift/library-go/pkg/route/secretmanager/fake"
+
+	routev1 "github.com/openshift/api/route/v1"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/watch"
+	testclient "k8s.io/client-go/kubernetes/fake"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+type testSARCreator struct {
+	allow bool
+	err   error
+	sar   *authorizationv1.SubjectAccessReview
+}
+
+func (t *testSARCreator) Create(_ context.Context, subjectAccessReview *authorizationv1.SubjectAccessReview, _ metav1.CreateOptions) (*authorizationv1.SubjectAccessReview, error) {
+	t.sar = subjectAccessReview
+	return &authorizationv1.SubjectAccessReview{
+		Status: authorizationv1.SubjectAccessReviewStatus{
+			Allowed: t.allow,
+		},
+	}, t.err
+}
+
+type testSecretGetter struct {
+	namespace string
+	secret    *corev1.Secret
+}
+
+func (t *testSecretGetter) Secrets(_ string) corev1client.SecretInterface {
+	return testclient.NewSimpleClientset(t.secret).CoreV1().Secrets(t.namespace)
+}
+
+// fakeSecretInformer will list/watch only one secret inside a namespace
+func fakeSecretInformer(ctx context.Context, fakeKubeClient *testclient.Clientset, namespace, name string) cache.SharedInformer {
+	fieldSelector := fields.OneTermEqualSelector("metadata.name", name).String()
+	klog.Info(fieldSelector)
```

Comment:
Why this log in a fake function used only in tests?

Reply:
Removed, thanks